### ISSUE TITLE
About simulation time

### ISSUE THREAD

**Issue by MontaEllis** [ipc-sim/Codim-IPC#17]
Great job! I'm curious, how long will the code take to run? I've been running the garment simulation code for about half a day, and it's still going!

**Comment by liminchen** [comment]
Thanks! 
Most examples should finish running in one or several hours. See timing table here:
<img width="1136" alt="Screen Shot 2024-07-03 at 09 41 12" src="https://github.com/ipc-sim/Codim-IPC/assets/6930359/279f9821-b71e-4c55-9e79-2cb26094746f">
https://github.com/KemengHuang/GPU_IPC has an efficient GPU implementation of our simulator, and it shuold be orders-of-magnitude faster.

**Comment by MontaEllis** [comment]
It appears that there is no Python version...

**Comment by MontaEllis** [comment]
It seems like the 5_garments demo will take a lot of time.

**Comment by liminchen** [comment]
Is it getting stuck at a specific frame or just generally slow? Our code is only tested on Ubuntu with MKL compiled CHOLMOD, so using different environment may also cause performance differences.

**Comment by MontaEllis** [comment]
How can I tell if it's stuck or not? I'm new to this and not very familiar with simulations. Your repository is my guide. I'm still learning and trying my best.

**Comment by liminchen** [comment]
So the simulator will output an obj file for the result of each frame, and you can take a look at the files, if the last output file was a long time ago, much longer than the adjacent time interval between earlier output files, then it means it get stuck at this frame.

You could also take screenshot of the command line output here.

If you are new to simulation, I guess you may find https://phys-sim-book.github.io/ and https://github.com/phys-sim-book/solid-sim-tutorial helpful.

**Comment by MontaEllis** [comment]
Thank you! I will read the physics simulation book carefully. Can I leave this issue open to ask more questions about garment simulation later?

**Comment by MontaEllis** [comment]
Actually, I'm trying to reimplement PhysAvatar, a nice project using your great C-IPC.

**Comment by liminchen** [comment]
Great! You can feel free to email me if you have followup questions. But note that sometimes I may be slow to reply :)

**Comment by MontaEllis** [comment]
Should we determine the pinned vertices ourselves in the garment demo?

**Comment by liminchen** [comment]
Yes, seams should be part of the input file, and dirichlet BC is set via python. Please see the example.

**Comment by MontaEllis** [comment]
When I run the code, the initial results seem very strange. Could you provide some advice?

input: 
![image](https://github.com/user-attachments/assets/be39fb79-8177-407c-9877-3b9829c26728)

![image](https://github.com/user-attachments/assets/9a812fe7-2c74-4fa2-8935-889baaf495b8)


results:
![image](https://github.com/user-attachments/assets/2198b5ce-982b-456a-911f-07f65e923ec7)

**Comment by MontaEllis** [comment]
Also, I'm curious why the drape and stage are vertical?

**Comment by MontaEllis** [comment]
I believe I followed the same structure as the demo, but the face might be incorrect. How should the face pattern and body be combined?

![image](https://github.com/user-attachments/assets/fac43e44-029a-46bd-b250-da7e8a5b8b6d)

**Comment by MontaEllis** [comment]
How do you achieve seams like this? I can't load it in Blender.

![image](https://github.com/user-attachments/assets/f0e24af1-b6dc-49b8-a3fd-33fb17345eb6)
